app/Example.java

```diff
@@ -0,0 +1,50 @@
+import android.view.View;
+
+import io.ably.lib.realtime.AblyRealtime;
+import io.ably.lib.realtime.Channel;
+import io.ably.lib.types.AblyException;
+import io.ably.lib.types.Message;
+import io.ably.lib.types.Param;
+
+public class Example {
+
+	private static final String API_KEY = "INSERT-YOUR-API-KEY-HERE";
+
+	public void publishAndSubcscribe(Button buttonPublish, Button buttonHistory) {
+		AblyRealtime realtime = new AblyRealtime(API_KEY);
+        /* Get a channel in the "persisted" namespace */
+        final Channel channel = realtime.channels.get("persisted:sounds");
+
+        /* Publish three messages when the publish button is clicked */
+        buttonPublish.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                /* Surround Channel#publish with try/catch */
+                try {
+                    channel.publish("play", "bark");
```

Comment:
Same as publishing, might be nice to have some feedback to show that something has happened i.e. publishign messages and then publish complete etc.  See https://github.com/ably/tutorials/blob/history-ruby/example.rb#L9 for an example.  As this is event driven we should confirm when that action is done i.e. publishing done